great_expectations/rule_based_profiler/domain_builder/cardinality_checker.py

```diff
@@ -0,0 +1,211 @@
+import abc
+import enum
+from dataclasses import dataclass
+from typing import Optional, Tuple, Union
+
+from great_expectations.exceptions import ProfilerConfigurationError
+
+
+@dataclass
+class CardinalityLimit(abc.ABC):
+    name: str
+
+
+@dataclass
+class RelativeCardinalityLimit(CardinalityLimit):
+    max_proportion_unique: Union[int, float]
+    related_metric_name: str = "column.unique_proportion"
+
+
+@dataclass
+class AbsoluteCardinalityLimit(CardinalityLimit):
+    max_unique_values: int
+    related_metric_name: str = "column.distinct_values.count"
+
+
+class CardinalityLimitMode(enum.Enum):
+    """Preset limits based on unique values (cardinality)
+
+    Defines relative (ratio) and absolute number of records (table rows) that
+    correspond to each cardinality category.
+
+    Used to determine appropriate Expectation configurations based on data.
+    """
+
+    ZERO = AbsoluteCardinalityLimit("ZERO", 0)
```

Comment:
@anthonyburdi Type hints for each, perhaps?.. not sure if overkill (I put them in for the profiler reconciliation strategy).  Thanks.

Reply:
Feels like overkill in this particular enum (assuming you mean e.g. `ZERO: AbsoluteCardinalityLimit = AbsoluteCardinalityLimit("ZERO", 0)`). If we want 100% type coverage though I agree then we will have to do it.

Reply:
@anthonyburdi Instead of my preference, I will trust @cdkini 's intuition for what is sufficient to do here.  Thanks!